Controllers/FileController.cs

```diff
@@ -1,49 +1,220 @@
+    public class FormData
+    {
+        public string[] Files { get; set; }
+        public Dictionary<string, int> DessertVotes { get; set; }
+        public string CurrentWeather { get; set; }
+        public List<Author> AuthorsList { get; set; }
+    }
+    public class AppSettings
+    {
+        public string UploadPath { get; set; }
+    }
+
     [Route("api/[controller]")]
     [ApiController]
     public class FileController : ControllerBase
     {
+        private string uploadPath;
+        private const string FilePath = "formDataStorage.json";
+        private Dictionary<string, int> dessertVotes = new Dictionary<string, int>();
+        private string? currentWeather = "clear";
+        private List<Author> authorsList = new List<Author> {};
+        private FormData allData;
+        
+        private FormData CreateFormData()
+        {
+            return new FormData
+            {
+                Files = Directory.GetFiles(uploadPath).Select(e => Path.GetFileName(e)).ToArray(),
+                DessertVotes = dessertVotes,
+                CurrentWeather = currentWeather ?? "clear",
+                AuthorsList = authorsList
+            };
+        }
+        private async Task InitializeFormTypes()
+        {
+            dessertVotes = new Dictionary<string, int>()
+            {
+                {"Red Velvet Cake", 0},
+                {"Peach Pie", 0},
+                {"Lemon Bars", 0}
+            };
+            authorsList = new List<Author>();
+            currentWeather = "clear";
+
+            if (System.IO.File.Exists(FilePath))
+            {
+                var json = await System.IO.File.ReadAllTextAsync(FilePath);
+                Console.WriteLine(json);
+                allData = JsonSerializer.Deserialize<FormData>(json);
+                dessertVotes = allData?.DessertVotes ?? dessertVotes;
+                authorsList = allData?.AuthorsList ?? authorsList;
+                currentWeather = allData?.CurrentWeather ?? currentWeather;
+            } 
+            else 
+            {
+                allData = new FormData
+                {
+                    Files = new string[] {},
+                    DessertVotes = dessertVotes,
+                    CurrentWeather = currentWeather,
+                    AuthorsList = authorsList
+                };
+                var json = JsonSerializer.Serialize(allData);
+                await System.IO.File.WriteAllTextAsync(FilePath, json);
+            }
+        }
+        private void IncrementDessertVote(string dessert) {
+            if (this.dessertVotes.ContainsKey(dessert))
+            {
+                this.dessertVotes[dessert] += 1;
+            }
+            else
+            {
+                this.dessertVotes[dessert] = 1;
+            }
+        }
+        public FileController(IOptions<AppSettings> settings)
+        {
+            uploadPath = string.IsNullOrEmpty(settings?.Value?.UploadPath) 
+                ? "wwwroot/upload" 
```

Comment:
Putting files in wwwroot makes them browsable, I'm not sure this is what we want here. We could just use an upload directory instead, however.